clientconn.go

```diff
@@ -924,12 +918,18 @@ func (ac *addrConn) connect() error {
 	return nil
 }
 
+func equalAddress(a, b resolver.Address) bool {
+	return a.Addr == b.Addr && a.ServerName == b.ServerName &&
+		a.Attributes.Equal(b.Attributes) &&
+		a.Metadata == b.Metadata
+}
+
 func equalAddresses(a, b []resolver.Address) bool {
 	if len(a) != len(b) {
 		return false
 	}
 	for i, v := range a {
-		if !v.Equal(b[i]) {
+		if !equalAddress(v, b[i]) {
 			return false
 		}
 	}
```

Comment:
This could be replaced with a call to:
 `slices.Equal(a, b, func(a, b resolver.Address) bool { return equalAddress(a, b)})`

Reply:
Done.